projects/user_publication/gscholar.py

```diff
@@ -0,0 +1,207 @@
+        return inner_f
+
+    def _publication_id(self, pub: dict):
+        m = re.search(r"cites=[\d+,]*", pub["citedby_url"])
+        return m.group()[6:]
+
+    def new_proxy(self):
+        """sets a new proxy from free-proxy library"""
+        if not self.pg.proxy_mode.value == "FREE_PROXIES":
+            return
+        logger.info("Getting new free proxy")
+        self.pg.get_next_proxy()
+        self.scholarly.use_proxy(self.pg)
+
+    @_handle_proxy_reload
+    def get_one_pub(self, title: str):
+        """To get one publication from google scholar
+        Make sure the title is an exact match
+
+        Args:
+            title (str): Title of the publication
+        """
+        return self.scholarly.search_single_pub(title)
+
+    @_handle_proxy_reload
+    def get_cites(
+            self, pub, year_low=2014,
+            year_high=datetime.now().year
+    ):
+        """Returns all the citations of the publication
+
+        Args:
+            pub (dict): return of the search_single_pub method
+            year_low (int, optional): year to query from. Defaults to 2014
+            year_high (int, optional): yaar to query till. Defaults to datetime.now().year
+
+        Returns:
+            list: list of publications that cited arg:pub
+        """
+        pub_id = self._publication_id(pub)
+        logger.info(f"Getting citations for {pub['bib']['title']}")
+        no_citations = pub['num_citations']
+        cites_gen = self.scholarly.search_citedby(
+            pub_id, year_low=year_low, year_high=year_high
+        )
+        citations = []
+        cit_count = 0
+        while True:
+            try:
+                cited_pub = self._get_bib(self._get_next_cite(cites_gen))
+                citations.append(cited_pub)
+                cit_count += 1
+                logger.info(f"Got {len(cit_count)} / {no_citations}")
+            except StopIteration:
+                logger.info(f"End of iteration. Got {len(citations)} / {no_citations}")
+                return citations
+
+    @_handle_proxy_reload
+    def _get_next_cite(self, cites_gen):
+        return next(cites_gen)
+
+    @_handle_proxy_reload
+    def _get_bib(self, pub):
+        bibt = self.scholarly.bibtex(pub)
+        pub.update({"bibtex": bibt})
+        return pub
+
+    def get_authors(self, pub):
+        """returns a list of authors in a publication
+
+        Args:
+            pub (dict): scholarly return of the publication
+
+        Returns:
+            list: list of authors in ["firstname lastname",] format
+        """
+        bibt = pub['bibtex']
+        bibt = utils.decode_unicode_text(bibt)
+        match = re.search(r'author\s*=\s*\{([\w\s-,]+)\}', bibt)
+        if match:
+            authors_t = match.groups()[0]
+        else:
+            # this mostly never happens - if it does - log it
+            logger.info(f"cannot parse bibtex, so ignoring - {bibt}")
+            return
+        if "and" in authors_t:

```

Comment:
You don't need to check for this. Running `authors_t.split(" and" )` will return the same result as what you've done in your else statement if `"and"` does not appear in the string. Also, what does the `t` in `authors_t` mean? There's no need to shorten variable names so much. It's better to be clear and unambiguous for future readers.

Reply:
Right. it means ```authors_text``` will change it to ```authors_match```